further_link/__main__.py

```diff
@@ -65,18 +137,34 @@ async def version(_):
     return app
 
 
+def make_synchronous(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        return asyncio.run(func(*args, **kwargs))
+
+    return wrapper
+
+
+async def create_app():
+    await create_bluetooth_app()
+
+    app = await create_web_app()
+    # Default handle_signals=True will ignore sigterm signals whilst there
+    # are requests that are not complete. This isn't appropriate for our
+    # indefinitely running websockets and can cause device shutdown to hang
+    runner = web.AppRunner(app, handle_signals=False)
+    await runner.setup()
+    site = web.TCPSite(runner, port=port(), ssl_context=ssl_context())
+    await site.start()
+
+
 @click.command()
-def main():
+@make_synchronous
+async def main():
     vnc.create_ssl_certificate()
-    return web.run_app(
-        create_app(),
-        port=port(),
-        ssl_context=ssl_context(),
-        # Default handle_signals=True will ignore sigterm signals whilst there
-        # are requests that are not complete. This isn't appropriate for our
-        # indefinitely running websockets and can cause device shutdown to hang
-        handle_signals=False,
-    )
```

Comment:
maybe we can keep this sync way of starting the web app and just before it start the bluetooth app using the make_sync decorator, or calling create_bt_app from inside create_app.

whatever ends up simpler is basically the way to go. probably best to avoid the tcpsite stuff because it's sort of an itnernal api and if we can avoid the make_sync decorator that would also be nice as it looks a bit confusing